nwb_conversion_tools/utils/ephys_writer.py

```diff
@@ -0,0 +1,99 @@
+from distutils.version import StrictVersion
+import pynwb
+from pathlib import Path
+from jsonschema import validate
+
+# import ephys writers
+from .si013nwbephyswriter import SI013NwbEphysWriter
+from .si090nwbephyswriter import SI090NwbEphysWriter
+from .neonwbephyswriter import NEONwbEphysWriter
+from nwb_conversion_tools.nwbconverter import NWBConverter
+from .common_writer_tools import default_export_ops, default_export_ops_schema
+
+
+def map_si_object_to_writer(object_to_write):
+    writer_class = None
+    try:
+        if isinstance(object_to_write, SI090NwbEphysWriter.supported_types()):
+            writer_class = SI090NwbEphysWriter
+    except AssertionError:
+        pass
+    try:
+        if isinstance(object_to_write, SI013NwbEphysWriter.supported_types()):
+            writer_class = SI013NwbEphysWriter
+    except AssertionError:
+        pass
+    try:
+        if isinstance(object_to_write, NEONwbEphysWriter.supported_types()):
+            writer_class = NEONwbEphysWriter
+    except AssertionError:
+        pass
+    if writer_class is None:
+        raise Exception(f"Could not write object of type {type(object_to_write)}")
+    return writer_class
+
+
+def export_ecephys_to_nwb(
+    object_to_write,
+    nwb_file_path=None,
+    nwbfile=None,
+    metadata=None,
+    **kwargs,
+):
+    """
+    Writes one object to NWB.
+
+    Supported objects are:
+        - spikeinterface version <= 0.13 extractors
+          (spikeextractors.RecordingExtractor / spikeextractors.SortingExtractor)
+        - spikeinterface version >= 0.90 objects
+          (spikeinterface.RecordingExtractor / spikeinterface.SortingExtractor / spikeinterface.WaveformExtractor)
+        - NEO >= 0.10 objects
+          (neo.rawIO / neo.IO)
+
+    Parameters
+    ----------
+    object_to_write
+    nwb_file_path
+    metadata
+    kwargs
```

Comment:
@Saksham20 can you add docstrings here? Especially for kwargs

Reply:
https://github.com/catalystneuro/nwb-conversion-tools/pull/230/commits/eaa4691d053025d78ca12738f827a6aa31677172